packages/trader/src/tsconfig.json

```diff
@@ -0,0 +1,9 @@
+{
```

Comment:
There are now two typescript "projects". This seems to be the new recommended setup for separating test and prod code in ts.

Effectively, everything in src is considered a separate project. It can be compiled on its own, but it's generally compiled as a dependency of the outer config that exists in the top level trader directory.

This has some performance enhancements, but the main benefit is that we can have a separate output dir -- `dist`. Hardhat automatically uses the tsconfig.json at the top level for testing (they don't let you point it to a different filename, which was the setup we had before), so that config _must_ build the test files. The output in that project, which are _distinct_ from those in this project, will be dumped in `build`. That will include the compiled hardhat config and all the compiled test files, etc.

Reply:
i think i get this. so theres a tsconfig in src which builds bot independently from tests, then there is a tsconfig in root which builds tests as well as src? If this is the case, what are your thoughts about keeping them _more_ seperate? ie tests and src contain their own tsconfigs and they are siblings rather parent child? 

I guess my thought process is i would rather not treat tests as a special case (root tsconfig), and use the same abstraction you did for src, treat it as its own project.

Reply:
The issue is that I believe the hardhat config needs to be in the same directory where 1) you run hardhat and 2) the test tsconfig.json is. So unless we always want to run hardhat from test/, we need to have the test tsconfig at the root of the package. Thoughts?

Reply:
@daywiss gonna go ahead and merge because I don't want to block @chrismaree, but happy to modify this setup in a follow-up!